### PR TITLE
TMEDIA-197 - Extract common components from promos

### PR DESCRIPTION
**Pull request by matthewroach**
## Description

Made reusable components and implemented reusable component into promo blocks

## Jira Ticket
- [TMEDIA-197](https://arcpublishing.atlassian.net/browse/TMEDIA-197)

## Acceptance Criteria
* Extract reused code from promos into shared styles and update promo blocks to use shared components

## Test Steps

1. Checkout this branch `git checkout TMEIDA-197-extract-common-components-from-promos`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/extra-large-promo-block,@wpmedia/large-promo-block,@wpmedia/medium-promo-block,@wpmedia/small-promo-block,@wpmedia/extra-large-manual-promo-block,@wpmedia/large-manual-promo-block,@wpmedia/medium-manual-promo-block,@wpmedia/small-manual-promo-block,@wpmedia/shared-styles`
3.  Verify promo blocks didn't change - test page - http://localhost/pf/test-all-promos/?_website=the-gazette

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [X] Confirmed all the test steps a reviewer will follow above are working.
- [X] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [X] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [X] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [X] Confirmed this PR has unit test files
  - [X] Ran `npm run test`, made sure all tests are passing
- [X] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/extra-large-manual-promo-block/features/extra-large-manual-promo/default.jsx [hunk 1]
```diff
@@ -38,25 +22,14 @@ const ExtraLargeManualPromoItem = ({ customFields }) => {
...
-              {((customFields.showOverline && customFields.overline) && !customFields.overlineURL)
-              && (
-                <OverlineHeader
-                  primaryFont={getThemeStyle(arcSite)['primary-font-family']}
-                  className="overline"
-                >
-                  {customFields.overline}
-                </OverlineHeader>
-              )}
+              {(customFields.showOverline)
```
Comment (JackHowa) [inline blocks/extra-large-manual-promo-block/features/extra-large-manual-promo/default.jsx:25]: nice

File: blocks/extra-large-promo-block/features/extra-large-promo/default.jsx [hunk 2]
```diff
@@ -120,73 +113,6 @@ const ExtraLargePromoItem = ({ customFields }) => {
     }`,
   }) || null;
 
-  const { website_section: websiteSection } = content?.websites?.[arcSite] ?? {
-    website_section: null,
-  };
-  const descriptionText = content && content.description ? content.description.basic : null;
```
Comment (JackHowa) [inline blocks/extra-large-promo-block/features/extra-large-promo/default.jsx:126]: so happy this is out of each block 👍

File: blocks/shared-styles/_children/.npmignore [hunk 3]
```diff
@@ -1 +1,2 @@
-*.test.jsx
\ No newline at end of file
+*.test.jsx
+*.story.*
```
Comment (JackHowa) [inline blocks/shared-styles/_children/.npmignore:2]: good catch

File: blocks/shared-styles/_children/byline/index.scss [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    display: inline;
+    font-size: 0.875rem;
+    line-height: 1rem;
+
+    #{$self}__by {
+      color: #3b3b3b;
+      margin-right: 0;
+    }
+
+    #{$self}__names {
```
Comment (JackHowa) [inline blocks/shared-styles/_children/byline/index.scss:17]: nice, this $self new?

File: blocks/shared-styles/_children/byline/index.scss [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  font-size: 1.25rem;
+  line-height: 2.5rem;
+
+  &--list {
+    display: inline;
+    font-size: 0.875rem;
+    line-height: 1rem;
+
+    #{$self}__by {
+      color: #3b3b3b;
```
Comment (JackHowa) [inline blocks/shared-styles/_children/byline/index.scss:13]: not sure if this is new but can we use arc variables for colors if possible? I believe they're in the news theme css

File: blocks/shared-styles/_children/byline/index.story.jsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+      },
+    },
+  };
+
+  return (
+    <Byline {...props} />
+  );
+};
+
+export const noAuthors = () => {
```
Comment (JackHowa) [inline blocks/shared-styles/_children/byline/index.story.jsx:69]: boom, nice work. this fixes https://arcpublishing.atlassian.net/browse/TMEDIA-208 I believe?

File: blocks/shared-styles/_children/promo-date/index.jsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  const {
+    dateLocalization: { language, timeZone, dateTimeFormat } = { language: 'en', timeZone: 'GMT', dateFormat: 'LLLL d, yyyy \'at\' K:m bbbb z' },
+  } = getProperties(arcSite);
+
+  const displayDate = content?.display_date || date;
+  const formattedDate = (displayDate && Date.parse(displayDate)) ? localizeDateTime(new Date(displayDate), dateTimeFormat, language, timeZone) : '';
+
+  return displayDate ? (
+    <PrimaryFont
+      as="time"
```
Comment (JackHowa) [inline blocks/shared-styles/_children/promo-date/index.jsx:25]: nice work

File: jest.config.js [hunk 8]
```diff
@@ -36,5 +36,6 @@ module.exports = {
     '!**/vendor/**',
     '!**/images/*.svg',
     '!**/mock*.js',
+    '!**/*.story*.jsx',
```
Comment (JackHowa) [inline jest.config.js:39]: good good

**Review by JackHowa — CHANGES_REQUESTED** [review]
Mostly looks good. But I noticed that we might be missing bold or font size based on comparing prod with local 
Screen Shot 2021-04-16 at 09 48 49

File: blocks/extra-large-manual-promo-block/features/extra-large-manual-promo/default.jsx [hunk 9]
```diff
@@ -1,33 +1,17 @@
...
 
 import '@wpmedia/shared-styles/scss/_extra-large-promo.scss';
 
-const DescriptionText = styled.p`
-  font-family: ${(props) => props.secondaryFont};
-`;
-
-const OverlineLink = styled.a`
-  font-family: ${(props) => props.primaryFont};
-  font-weight: bold;
```
Comment (JackHowa) [inline blocks/extra-large-manual-promo-block/features/extra-large-manual-promo/default.jsx:19]: might be missing this font-weight here?

**Comment by JackHowa** [comment]
also, it kind of seems like description font family was not working previously in stable? Good that it should be working now. looks serif in prod and sans-serif in your local (same site gazette)
Screen Shot 2021-04-16 at 09 50 59

File: blocks/shared-styles/_children/byline/index.scss (hunk 4 above)
Comment (matthewroach) [reply blocks/shared-styles/_children/byline/index.scss:17]: No, it's a Sass way to do BEM naming with nesting

File: blocks/shared-styles/_children/byline/index.story.jsx (hunk 6 above)
Comment (matthewroach) [reply blocks/shared-styles/_children/byline/index.story.jsx:69]: Yes, should do

File: blocks/shared-styles/_children/byline/index.scss (hunk 5 above)
Comment (matthewroach) [reply blocks/shared-styles/_children/byline/index.scss:13]: This color value isn't in news-theme-css.. We could create a ticket for this - I just brought over the existing styled component logic

File: blocks/extra-large-manual-promo-block/features/extra-large-manual-promo/default.jsx (hunk 9 above)
Comment (matthewroach) [reply blocks/extra-large-manual-promo-block/features/extra-large-manual-promo/default.jsx:19]: It's in the Sass file now

**Comment by matthewroach** [comment]
@JackHowa Looks like the font settings are different with the current prod bundle to what is in Fusion-News-Theme repo and why you are seeing differences

**Review by JackHowa — APPROVED** [review]
Looks good based on comparing with prod. font style is different in prod than what I have locally

File: blocks/shared-styles/_children/byline/index.scss (hunk 5 above)
Comment (JackHowa) [reply blocks/shared-styles/_children/byline/index.scss:13]: ok no worries, thought this was new

**Comment by JackHowa** [comment]
Screen Shot 2021-04-16 at 13 34 52

@matthewroach verified with prod https://corecomponents.arcpublishing.com/pf/test-all-promos/?_website=the-gazette&d=751

**Pull request merged**